src/components/PupilJourneyComponents/QuizEngineProvider/QuizEngineProvider.tsx

```diff
@@ -50,95 +56,113 @@ export const QuizEngineProvider = memo((props: QuizEngineProps) => {
     [questionsArray],
   );
 
-  const numberOfQuestions = useMemo(
-    () => multipleChoiceQuestionsArray.length,
-    [multipleChoiceQuestionsArray],
-  );
-
   const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
   const [currentQuestionData, setCurrentQuestionData] = useState(
-    multipleChoiceQuestionsArray[currentQuestionIndex],
+    filteredQuestions[currentQuestionIndex],
+  );
+  const [questionState, setQuestionState] = useState<QuestionState[]>(
+    filteredQuestions.map(() => ({
+      mode: "init",
+      offerHint: false,
+      grade: 0,
+      feedback: undefined,
+    })),
   );
-  const [questionState, setQuestionState] = useState<QuestionState>({
-    mode: "input",
-    answer: undefined,
-    offerHint: false,
-    score: 0,
-    maximumScore: numberOfQuestions,
-  });
+  const [score, setScore] = useState(0);
+  const maxScore = useMemo(() => filteredQuestions.length, [filteredQuestions]);
+  const [isComplete, setIsComplete] = useState(false);
 
   useEffect(() => {
-    const desiredQuestionData =
-      multipleChoiceQuestionsArray[currentQuestionIndex];
-    setCurrentQuestionData(desiredQuestionData);
-  }, [currentQuestionIndex, multipleChoiceQuestionsArray]);
+    setCurrentQuestionData(filteredQuestions[currentQuestionIndex]);
+  }, [currentQuestionIndex, filteredQuestions]);
```

Comment:
This `useEffect` doesn't seem necessary. You can ditch the accompanying `useState` too and always point to the item in `filteredQuestions`.

```typescript
const currentQuestionData = filteredQuestions[currentQuestionIndex];
```

and spare yourself the extra render of this component caused by `useEffect`